build-system/tasks/serve.js

```diff
@@ -42,10 +42,30 @@ const {log} = require('../common/logging');
 const {watchDebounceDelay} = require('./helpers');
 const {watch} = require('chokidar');
 
+/**
+ * @typedef {{
+ *   name: string,
+ *   port: string,
+ *   root: string,
+ *   host: string,
+ *   debug?: boolean,
+ *   silent?: boolean,
+ *   https?: boolean,
+ *   preferHttp1?: boolean,
+ *   liveReload?: boolean,
+ *   middleware?: function[],
+ *   startedcallback?: function,
+ *   serverInit?: function,
+ *   fallback?: string,
+ *   index: boolean | string | string[],
+ * }}
+ */
+let GulpConnectOptions; // eslint-disable-line no-unused-vars
```

Comment:
Our lint checker allows unused vars with certain names:

https://github.com/ampproject/amphtml/blob/162d7544f7246c26bcd012c443225c53c60a455c/.eslintrc.js#L207-L213

If you rename this typedef to `GulpConnectOptionsDef`, you won't need the `eslint-disable` annotation. (Might be able to do this in other places in `build-system/` too.)

Reply:
Oh wow, that is fantastic! I'll change the name of this typedef and then explore changing others in a separate PR.